extensions/content-manager/admin/src/containers/ListView/index.js

```diff
@@ -0,0 +1,584 @@
+      target: {
+        name,
+        slug,
+        value: !value,
+      },
+    });
+  };
+
+  const handleChangeParams = ({ target: { name, value } }) => {
+    const updatedSearch = getSearchParams({ [name]: value });
+    const newSearch = generateSearchFromFilters(updatedSearch);
+
+    if (name === '_limit') {
+      emitEvent('willChangeNumberOfEntriesPerPage');
+    }
+
+    push({ search: newSearch });
+    resetProps();
+    getDataRef.current(slug, updatedSearch);
+  };
+  const handleClickDelete = id => {
+    setIdToDelete(id);
+    toggleModalDelete();
+  };
+  const handleSubmit = (filters = []) => {
+    emitEvent('didFilterEntries');
+    toggleFilterPickerState();
+    handleChangeParams({ target: { name: 'filters', value: filters } });
+  };
+
+  const filterPickerActions = [
+    {
+      label: `${pluginId}.components.FiltersPickWrapper.PluginHeader.actions.clearAll`,
+      kind: 'secondary',
+      onClick: () => {
+        toggleFilterPickerState();
+        handleChangeParams({ target: { name: 'filters', value: [] } });
+      },
+    },
+    {
+      label: `${pluginId}.components.FiltersPickWrapper.PluginHeader.actions.apply`,
+      kind: 'primary',
+      type: 'submit',
+    },
+  ];
+
+  const headerAction = [
+    {
+      label: formatMessage(
+        {
+          id: 'content-manager.containers.List.addAnEntry',
+        },
+        {
+          entity: getName() || 'Content Manager',
+        }
+      ),
+      onClick: () => {
+        emitEvent('willCreateEntry');
+        push({
+          pathname: `${pathname}/create`,
+          search: `redirectUrl=${pathname}${search}`,
+        });
+      },
+      color: 'primary',
+      type: 'button',
+      icon: true,
+      style: {
+        paddingLeft: 15,
+        paddingRight: 15,
+        fontWeight: 600,
+      },
+    },
+  ];
+
+  const headerProps = {
+    title: {
+      label: getName() || 'Content Manager',
+    },
+    content: formatMessage(
+      {
+        id:
+        count > 1
+        ? `${pluginId}.containers.List.pluginHeaderDescription`
+        : `${pluginId}.containers.List.pluginHeaderDescription.singular`,
+      },
+      { label: count }
+    ),
+    actions: headerAction,
+  };
+
+  const relation = layouts[slug].contentType.settings.relationName;
+  const relationColumnName = layouts[slug].contentType.settings.relationKey;
+
+  const getIdFromParentAndChild = (obj, idsContainter) => {
+    if(idsContainter.includes(obj.id)) {
+      idsContainter.push(obj.id);
+      return;
+    }
+    idsContainter.push(obj.id);
+    if(obj.child) {
+      getIdFromParentAndChild(obj.child, idsContainter);
+    }
+    if(obj[relation] && obj[relation].id === obj.id){
+      idsContainter.push(obj.id)
+    }
+  }
+
+  const collectIdsFromNestedData = (nestedData) => {
+    let ids = [];
+    nestedData.forEach(el => {
+      getIdFromParentAndChild(el, ids)
+    })
+    return ids;
+  }
+
+  const createRelationStructure = () => {
```

Comment:
it seems to be core of this extension so:
1. extract to separate module
2. write unit test only for this functions
3. use as descriptive name as possible (`x`)

Reply:
Will do in next step with validation and tests